Make DataModel.update no longer update from extra_fits

jdavies-st opened the pull request:
DataModel.update() method will now ignore `extra_fits` in the model that is providing the updated metadata, unless requested via keyword arg `extra_fits=True`.

Resolves #4591

hbushouse commented on jwst/datamodels/model_base.py:
```diff
@@ -948,12 +948,6 @@ def protected_keyword(path):
             if not protected_keyword(path):
                 set_hdu_keyword(self._instance, d, path)
 
-        # Perform updates to extra_fits area of a model
```
I would really like to see this controlled by an optional argument, rather than just eliminating the capability.

jdavies-st commented:
One thing I'm not entirely clear on in the `update()` method API is what `only` arg does.  The documentation is bad, and the default value of `""` is, um, not good.  @hbushouse do you have any insight what its purpose is?  I feel like there could be an odd interaction between `only` and `extra_fits` args.

hbushouse commented:
IIRC, the "only" arg was used to copy/update only meta data from the primary header (no extensions).

jdavies-st commented:
I think the tests now should cover the combos of `only` and `extra_fits`.  And it seems to behave how we expect it to.

I also found a test for this method over in `test_fits.py`.  That test was in the wrong location and was very simple and not very intuitive.  I removed it, and added another test in `test_models.py`.  I think the new tests cover all the use cases and give good coverage to the codebase.
